### ISSUE TITLE
Should we depend on unwind.h being present in clang's builtin include path

### ISSUE THREAD

**Issue by sbc100** [emscripten-core/emscripten#9778]
As part of #9673 it was discovered that not all clang installations contain `unwind.h`.   Normally this lives in `lib/clang/10.0.0/include` alongside the clang install but at least on one BSD distribution this file gets stripped out.

Should we support besoke clang installations like this, or try to detect and error out in a nicer way at least?  Or we could just start shipping our own copy of this header?

**Comment by yurivict** [comment]
IMO, ship your own copy, It' safer this way, and an easy solution to the problem.

**Comment by kripken** [comment]
I think so too. Unless there's some risk - do these headers change much over time?

**Comment by yurivict** [comment]
unwind.h actually has nothing to do with llvm. It comes from the ```libunwind``` project https://www.nongnu.org/libunwind/, llvm absorbed its headers and source files.

**Comment by sbc100** [comment]
True, but its install this header in a clang-specific include location is part of the normal clang install.   I don't think you can say you have a full clang install without this header.   Can you point me to the BSD patch that removes this header?  It might shed some light on what motivated the removal.

**Comment by yurivict** [comment]
The patch is in the port's directory and it has no comment.
I asked the maintainer to re-add the file.

**Comment by sbc100** [comment]
Do you mean this things: https://github.com/freebsd/freebsd-ports/tree/master/devel/llvm-devel/?

Can you link to the patch?  The only one I can see there that look relevant is: https://github.com/freebsd/freebsd-ports/blob/master/devel/llvm-devel/files/clang/patch-tools_clang_lib_Headers_CMakeLists.txt.. which seems to exclude some headers but not `unwind.h`.  I must be looking at the wrong tree?

**Comment by sbc100** [comment]
Oh!  Its because they already fixed back in https://github.com/freebsd/freebsd-ports/commit/b58cbd9da2f1f1f917a826a3057d0013cca726d4.  Yay!

**Comment by yurivict** [comment]
Yes, they just re-added it recently.

**Comment by yurivict** [comment]
So this isn't a problem any more on FreeBSD.
